datafusion/physical-expr/src/aggregate/sum.rs

```diff
@@ -546,4 +555,76 @@ mod tests {
             Arc::new(Float64Array::from(vec![1_f64, 2_f64, 3_f64, 4_f64, 5_f64]));
         generic_test_op!(a, DataType::Float64, Sum, ScalarValue::from(15_f64))
     }
+
+    fn row_aggregate(
+        array: &ArrayRef,
+        agg: Arc<dyn AggregateExpr>,
+        row_accessor: &mut RowAccessor,
+    ) -> Result<ScalarValue> {
+        let mut accum = agg.create_row_accumulator(0)?;
+        let scalar_value = ScalarValue::try_from_array(array, 0)?;
+
+        accum.update_scalar(&scalar_value, row_accessor)?;
+        accum.evaluate(row_accessor)
+    }
+
+    #[test]
+    fn sum_dictionary_f64() -> Result<()> {
+        let keys = Int32Array::from(vec![0, 1, 2, 3, 4]);
```

Comment:
I recommend using keys that are different from values, to help improve the coverage.

Also perhaps a repeated value rather that distinct values would help

Reply:
Yea, improved the test coverage.